oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/TraceSamplingPolicyWatcher.java

```diff
@@ -0,0 +1,217 @@
+        if (map == null) {
+            return null;
+        }
+        SamplePolicySettings settings;
+        // default config
+        Object defaultMapObject = map.get("default");
+        if (defaultMapObject != null) {
+            Map<String, Object> defaultMap = (Map<String, Object>) defaultMapObject;
+            settings = new SamplePolicySettings(defaultMap.get("rate") == null ? null : (Integer) defaultMap.get("rate"),
+                    defaultMap.get("duration") == null ? null : (Integer) defaultMap.get("duration"));
+        } else {
+            settings = defaultSampleConfigSettings();
+        }
+        // services config
+        Object servicesObject = map.get("services");
+        if (servicesObject != null) {
+            List<Map<String, Object>> serviceList = (List<Map<String, Object>>) servicesObject;
+            serviceList.forEach(service -> {
+                String name = (String) service.get("name");
+                if (StringUtil.isBlank(name)) {
+                    return;
+                }
+                SamplePolicy samplePolicy = new SamplePolicy();
+                samplePolicy.setRate(service.get("rate") == null ? null : (Integer) service.get("rate"));
+                samplePolicy.setDuration(service.get("duration") == null ? null : (Integer) service.get("duration"));
+                settings.addPolicy(name, samplePolicy);
+            });
+        }
+        return settings;
+    }
+
+    private SamplePolicySettings defaultSampleConfigSettings() {
+        return new SamplePolicySettings(null, null);
+    }
+
+    @Getter
+    @Setter
+    @ToString
+    @AllArgsConstructor
+    @NoArgsConstructor
+    public static class SamplePolicy {
+        private Integer rate;
+        private Integer duration;
+    }
+
+    @ToString
+    public static class SamplePolicySettings {
+
+        @Getter
+        // `default` is the keyword ,so named `global`
+        private SamplePolicy global;
+        @Getter
+        private Map<String, SamplePolicy> services;
+
+        /**
+         * @param defaultRate     The sample rate precision is 1/10000. 10000 means 100% sample in default.
+         * @param defaultDuration Setting this threshold about the latency would make the slow trace segments sampled if they cost more time, even the sampling mechanism activated. The default value is `-1`, which means would not sample slow traces. Unit, millisecond.
+         */
+        SamplePolicySettings(Integer defaultRate, Integer defaultDuration) {
+            SamplePolicy samplePolicy = new SamplePolicy(defaultRate == null ? 10000 : defaultRate, defaultDuration == null ? -1 : defaultDuration);
+            this.global = samplePolicy;
+            this.services = new ConcurrentHashMap<>();
+        }
+
+        public void addPolicy(String service, SamplePolicy samplePolicy) {
+            this.services.put(service, samplePolicy);
+        }
+
+        public SamplePolicy getPolicy(String service) {
```

Comment:
This implementation is incorrect. It should be a simple `get`. You should have `service specific`->`global`->`hardcode default` logic here.